ml_dtypes/_src/dtypes.cc

```diff
@@ -60,6 +60,22 @@ struct TypeDescriptor<bfloat16> : CustomFloatType<bfloat16> {
   static constexpr char kNpyDescrByteorder = '=';
 };
 
+template <>
+struct TypeDescriptor<float8_e4m3> : CustomFloatType<float8_e4m3> {
+  typedef float8_e4m3 T;
+  static constexpr bool is_floating = true;
+  static constexpr bool is_integral = false;
+  static constexpr const char* kTypeName = "float8_e4m3";
+  static constexpr const char* kQualifiedTypeName = "ml_dtypes.float8_e4m3";
+  static constexpr const char* kTpDoc = "float8_e4m3 floating-point values";
+  // Treating e4m3 as the natural "float" type since it is IEEE-754 compliant.
```

Comment:
This may cause problems because e5m2 is also marked as the 1-byte float type.

(I'm not even sure this is really a problem now, since NumPy's promotion logic has been reworked significantly in recent years, but at least at some point having multiple float types with the same size confused things. We found that to be the case for bfloat16 and float16.)

That said, if we don't have a test case that shows it causes a problem, then I have no reason to object.

Reply:
- changed e4m3 kind to V (Void) and char = '7' (not used by any other np/ml_dtype types)
- updated changelog
- fixed lint errors